tests/test_utility.py

```diff
@@ -9,30 +9,35 @@
     "set_nodes, expected_nodes",
     [
         (
-            "https://firstnode.neurobagel.org/query",
-            ["https://firstnode.neurobagel.org/query/"],
+            "(https://firstnode.neurobagel.org/query, firstnode)",
+            {"https://firstnode.neurobagel.org/query/": "firstnode"},
         ),
         (
-            "https://firstnode.neurobagel.org/query/ https://secondnode.neurobagel.org/query",
-            [
-                "https://firstnode.neurobagel.org/query/",
-                "https://secondnode.neurobagel.org/query/",
-            ],
+            "(https://firstnode.neurobagel.org/query/, firstnode) (https://secondnode.neurobagel.org/query, secondnode)",
+            {
+                "https://firstnode.neurobagel.org/query/": "firstnode",
+                "https://secondnode.neurobagel.org/query/": "secondnode",
+            },
         ),
         (
-            " https://firstnode.neurobagel.org/query/ https://secondnode.neurobagel.org/query/  ",
-            [
-                "https://firstnode.neurobagel.org/query/",
-                "https://secondnode.neurobagel.org/query/",
-            ],
+            "(firstnode.neurobagel.org/query/, firstnode) (https://secondnode.neurobagel.org/query, secondnode)",
+            {
+                "https://secondnode.neurobagel.org/query/": "secondnode",
+            },
+        ),
+        (
+            "( , firstnode) (https://secondnode.neurobagel.org/query, secondnode)",
```

Comment:
I would split these cases off into a "bad input" type test and then both assert that the tool can process it (as now) but also that we raise a warning